### ISSUE TITLE
Make package activation more specific

### ISSUE THREAD

**Issue by corbett5** [LLNL/uberenv#64]
If you've installed multiple versions of say `py-scipy` you need to specify which one you want to activate, it would be nice if Uberenv could pick out the exact version that was used, or at least use the same compiler to pick out the right package.

```
==> lvarray: Executing phase: 'hostconfig'
[activating dependent packages]
[exe: spack/bin/spack activate py-scipy]
==> Error: py-scipy matches multiple packages.
  Matching packages:
    km2gnzw py-scipy@1.6.1%gcc@8.3.1 arch=linux-rhel7-x86_64
    db5qgqb py-scipy@1.6.1%clang@11.0.1 arch=linux-rhel7-x86_64
  Use a more specific spec.
```

**Comment by cyrush** [comment]
I agree this would be great, in the past we didn't have the infrastructure to do this -- but I think with the detailed spec info we now have (https://github.com/LLNL/uberenv/blob/b4ac7a1609a63565c3580cc0380c7d7ac56cb776/uberenv.py#L792), its probably  possible to make a table that maps the dependent package names to their correct hash.
